src/popup/router/pages/Retip.vue

```diff
@@ -62,25 +62,31 @@ export default {
       urlStatus(state, getters) {
         return getters['tipUrl/status'](this.tip.url);
       },
+
       validationStatus({ sdk }, { account, minTipAmount }) {
         if (!sdk || !this.tippingContract) {
           return { error: true };
         }
         if (this.selectedToken && this.$route.query.id.includes('_v1')) {
           return { error: true, msg: this.$t('pages.tipPage.v1FungibleTokenTipError') };
         }
-        if (!this.selectedToken && this.amount < minTipAmount) {
+        if (!+this.amount) {
+          return { error: true, msg: this.$t('pages.tipPage.requiredAmountError') };
+        }
+        if (!this.selectedToken && +this.amount < minTipAmount) {
           return { error: true, msg: this.$t('pages.tipPage.minAmountError') };
         }
         const fee = calculateFee(TX_TYPES.contractCall, {
           ...sdk.Ae.defaults,
           contractId: this.tippingContract.deployInfo.address,
           callerId: account.publicKey,
         });
+        const calculatedMaxValue = this.balance > fee ? this.balance.minus(fee) : 0;
         if (
           this.selectedToken
-            ? this.selectedToken.convertedBalance < this.amount || this.balance < fee
-            : this.balance < fee + this.amount
+            ? convertToken(this.selectedToken.balance, -this.selectedToken.decimals) <
+                this.parsedAmount || this.balance < fee
+            : calculatedMaxValue - this.parsedAmount <= 0
```

Comment:
I can't get the reasoning of this refactoring, the previous version was looking better. Did you try to do it more accurately?

Reply:
I did it more accurate, yes.